independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java

```diff
@@ -1414,6 +1462,12 @@ void implementCreateForClassBean(ClassOutput classOutput, ClassCreator beanCreat
                 create.invokeVirtualMethod(MethodDescriptor.of(methodInjection.target.asMethod()), instanceHandle,
                         referenceHandles);
             }
+
+            // Destroy injected transient references
+            for (TransientReference transientReference : transientReferences) {
```

Comment:
Maybe extract to a util method? The same destroy logic appears four times in this class.

Reply:
Well, it's just a very simple iteration and  the util method would have to accept three params. But you're probably right ;-)